src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -0,0 +1,172 @@
+package edu.kit.ipd.crowdcontrol.objectservice.moneytransfer;
+
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.GiftCodeRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.model.tables.records.WorkerRecord;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerBalanceOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.database.operations.WorkerOperations;
+import edu.kit.ipd.crowdcontrol.objectservice.mail.MailHandler;
+import edu.kit.ipd.crowdcontrol.objectservice.template.Template;
+import org.jooq.Result;
+
+import java.io.*;
+import java.util.*;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+
+/**
+ * Describes a Manager for money transfers. The Manager can log payments and pay off.
+ *
+ * @author Felix Rittler
+ */
+public class MoneyTransferManager {
+
+    MailHandler mailHandler;
+    WorkerBalanceOperations workerBalanceOperations;
+    WorkerOperations workerOperations;
+
+    int payOffThreshold;
+    int minGiftCodesCount;
+    String notificationMailAddress;
+    StringBuilder notificationText;
+
+
+    public MoneyTransferManager(MailHandler mailHandler, WorkerBalanceOperations workerBalanceOperations, WorkerOperations workerOperations, String notificationMailAddress) throws MessagingException {
+        this.mailHandler = mailHandler;
+        this.workerOperations = workerOperations;
+        this.workerBalanceOperations = workerBalanceOperations;
+        this.minGiftCodesCount = 10;
+        this.payOffThreshold = 0;
+        this.notificationMailAddress = notificationMailAddress;
+        this.notificationText = new StringBuilder();
+    }
+
+    /**
+     * Logs a new money transfer and saves it.
```

Comment:
it writes an outstanding balance to the db. You could use it as a synonym of record. The payment happens in the payOff() method